format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedInputFormatProvider.java

```diff
@@ -130,6 +130,9 @@ static Schema detectSchema(DelimitedConfig conf, String delimiter,
       }
       List<Schema.Field> fields = DataTypeDetectorUtils.detectDataTypeOfEachDatasetColumn(
         conf.getOverride(), columnNames, dataTypeDetectorStatusKeeper);
+      if (conf.getPathField() != null) {
```

Comment:
call addPathField() from the parent class

Reply:
Since detectSchema is a static method it cannot call non static method

Reply:
you can add a static method:

```
Schema addPathField(List<Schema.Field> dataFields, FailureCollector failureCollector) {
  return addPathField(conf, dataFields, failureCollector);
}

static Schema addPathField(PathTrackingConfig config, List<Schema.Field> dataFields,
                           FailureCollector failureCollector) {
  ...
}
```

It is important to make sure it is re-using the logic. By not calling the common code, this class now has the same problem as the others used to have, where a path field conflict will not be detected. Please also make sure you add unit tests to catch this edge case.

Reply:
Added static method